tsdate/core.py

```diff
@@ -858,6 +931,129 @@ def outside_maximization(self, *, eps, progress=None):
         return self.lik.timepoints[np.array(maximized_node_times).astype("int")]
 
 
+class ExpectationPropagation(InOutAlgorithms):
+    """
+    Expectation propagation, where the edge factors are approximated
```

Comment:
What is meant by "edge factors" here?

Reply:
Poisson likelihoods for edges -- worth writing out some terminology here